### PR TITLE
feat: #192 Add possibility to edit own way

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ */
+export const updateCell = (
+  text: string,
+  callback: (arg: boolean) => void,
+  arrayItem?: GoalPreview,
+  parentUuid?: string,
+  columnName?: keyof ColumnNameProps,
+) => {
+  if (arrayItem) {
+    updateCells(`update${arrayItem.constructor.name}`, text, arrayItem.uuid);
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts:30]: no template strings for parameters to update something, in is not maintainable

File: mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  return updateCellsFunctions[nameOfFunction](text, uuid);
+};
+
+/**
+ * Update cell value
+ */
+export const updateCell = (
+  text: string,
+  callback: (arg: boolean) => void,
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts:24]: what does `arg` mean?

Comment (tritonJS826) [reply mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts:24]: call back is not described meanin of this function

File: mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  arrayItem?: GoalPreview,
+  parentUuid?: string,
+  columnName?: keyof ColumnNameProps,
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateCell.ts:27]: wrong interface
there is no possibilities to have all this optional parameters

File: mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateGoalPreview.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import {GoalPreviewDAL} from "src/dataAccessLogic/GoalPreviewDAL";
+import {GoalPreview} from "src/model/businessModelPreview/GoalPreview";
+
+/**
+ * Update JobDone
+ */
+export const updateGoalPreview = async (description: string, uuid: string) => {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateGoalPreview.ts:7]: make param oblject

File: mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateName.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
+import {ColumnNameProps} from "src/logic/waysTable/renderCellValue/renderCellValue";
+import {WayPreview} from "src/model/businessModelPreview/WayPreview";
+
+/**
+ * Update Way
+ */
+export const updateName = async (updatedText: string, uuid: string, columnName: keyof ColumnNameProps) => {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/helpers/updateName.ts:8]: ?

File: mw-webapp/src/logic/waysTable/renderCellValue/renderCellValue.tsx [hunk 6]
```diff
@@ -1,10 +1,102 @@
...
+
+  /**
+   * Column Name
+   */
+  name: string;
+
+  /**
+   * Column isCompleted
+   */
+  isCompleted: boolean;
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/renderCellValue.tsx:20]: !

File: mw-webapp/src/logic/waysTable/renderCellValue/renderCellValue.tsx [hunk 7]
```diff
@@ -1,10 +1,102 @@
...
+  /**
+   * Value isCompleted for different styles
+   */
+  isWayCompleted?: boolean;
+}
+
+/**
+ * Render Input or span depend on client actions
+ */
+export const renderCellValue = (props: CellItemProps) => {
```
Comment (tritonJS826) [inline mw-webapp/src/logic/waysTable/renderCellValue/renderCellValue.tsx:57]: renderEditableText
why it is not a component

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by SergioAJS** [comment]
I've noticed that when I double-click on a value to change it, then exit edit mode without changing the value - firestore requests occur. Is it possible to make it so that before the request a check is performed to see if the value has changed, and if it hasn't, the request is not sent?

**Review by SergioAJS — COMMENTED** [review]
I've noticed that when I double-click on a value to change it, then exit edit mode without changing the value - firestore requests occur. Is it possible to make it so that before the request a check is performed to see if the value has changed, and if it hasn't, the request is not sent?

**Pull request closed without merging**